### PR TITLE
Adds Github actions [DRAFT] [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by theboy6545**
TODO: bother KMC to check the shit

## About The Pull Request

Replaces travis with github actions

## Why It's Good For The Game
So we can have bots checking our code without needing to pay people for it.

## Changelog
:cl:
add: Added new things
add: Added more things
del: Removed old things
tweak: tweaked a few things
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

**Comment by francinum** [comment]
You're going to need to port the ci scripts into the tools folder as well.

**Comment by theboy6545** [comment]
Debugging with the thing I'm debugging feels like having my arm inside a patent and asking what feels painful

File: tools/ci/run_server.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+tools/deploy.sh ci_test
+rm ci_test/*.dll
+mkdir ci_test/config
+
+#test config
+cp tools/ci/ci_config.txt ci_test/config/config.txt
+
+cd ci_test
+ln -s $HOME/libmariadb/libmariadb.so libmariadb.so
+DreamDaemon beestation.dmb -close -trusted -verbose -params "log-directory=ci" || EXIT_CODE=$?
```
Comment (drongood12) [inline tools/ci/run_server.sh:14]: beestation.dmb sure?

File: .dockerignore [hunk 2]
```diff
@@ -1,6 +1,5 @@
 .dockerignore
 .editorconfig
-.travis.yml
```
Comment (drongood12) [inline .dockerignore:3]: Why did you remove this if you have it at https://github.com/BeeStation/NSV13/blob/master/.dockerignore?

**Comment by drongood12** [comment]
It would be nice to add more .github/workflows/update_tgs_dmapi.yml, because this is a convenient workflow for those who use dmapi and tgs-server, so as not to manually update each time

File: .dockerignore (hunk 2 above)
Comment (theboy6545) [reply .dockerignore:3]: Because I forgot to remove it there, as I am a fool.

**Comment by Kmc2000** [comment]
Pull from master to get the “fix” to the thing

**Comment by theboy6545** [comment]
#1128 Is doing the same and it is being done by a competent person
